compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -3077,7 +3077,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         fields: &[Ident],
         expr: &'tcx hir::Expr<'tcx>,
     ) -> Ty<'tcx> {
-        let container = self.to_ty(container).normalized;
```

Comment:
I don't think we should be erasing regions this early in typeck... Why not erase later, like during writeback?

Reply:
Should I just replace this assert with the erase?

https://github.com/rust-lang/rust/blob/9052ca9393d3182b6c029d1d34b30837a3747f34/compiler/rustc_hir_typeck/src/writeback.rs#L696-L702

Reply:
Actually, we shouldn't be skipping inference here in the first place, because this leads to another error-to-ICE for this code:

```rust
#![feature(offset_of)]

struct Foo<T> {
    x: T,
}

fn main() {
    let _ = core::mem::offset_of!(Foo<_>, x);
}
```